New Cmdlets for Management.Sql to allow customers to add TDE keys and set TDE protector for managed instances

nivimsft opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

nivimsft committed:
Adding Swagger for APIs used to add ManagedInstance TDE keys and set TDE protector for ManagedInstances

nivimsft committed:
Adding ManagedInstanceKeys.json to readme.md

nivimsft committed:
Changes for ManagedInstanceEncryptionProtectors

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-go

AutorestCI commented:
# Automation for azure-sdk-for-node

nivimsft committed:
Removing Resource & ProxyResource and adding reference to "../../../common/v1/types.json#/definitions/ProxyResource"

jaredmoo commented on specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/ManagedInstanceEncryptionProtectors.json:
```diff
@@ -0,0 +1,355 @@
+          "ManagedInstanceEncryptionProtectors"
+        ],
+        "description": "Gets a list of managed instance encryption protectors",
+        "operationId": "ManagedInstanceEncryptionProtectors_ListByServer",
```
`ListByInstance`

nivimsft committed:
Changing operation id to ListByInstance as per Jared's recommendation

anuchandy reviewed (changes requested):
@nivimsft looks good. left some minor comments. please take look.

nivimsft committed:
Addressed comments By @anuchandy
- Changed comment to created or updated
- changed operation if to listByInstance

anuchandy commented:
@ravbhatnagar please review, the service is in production and was deployed on 08/26.
